### PR TITLE
CV3-47 Use celery to delete archived data that is over 30 days old

### PR DESCRIPTION
**Pull request by mifeille**
#### Description
Currently, the Backend implements soft deletes which instead of wiping a record from the database, changes the state of a row from active to archived. We would like to periodically remove rows whose state is archived from the database.

#### Type of change
Please select the relevant option
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others (cosmetics, styling, improvements)
- [ ] This change requires a documentation update

#### Checklist
- [x] My changes generate no new warnings
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [x] Existing unit tests pass locally with my changes
- [x] Implementation works according to expectations

#### JIRA
[CV3-47](https://andela-apprenticeship.atlassian.net/browse/CV3-47)

### CODE REVIEW

File: services/data_deletion/clean_archived_data.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from helpers.database import database_uri
+from sqlalchemy import create_engine, MetaData, and_
+from datetime import timedelta, datetime
+import celery
+
+
+@celery.task(name='clean_archived_data.delete_archived_data')
+def delete_archived_data():
+    database_engine = create_engine(database_uri)
```
Comment (MCFrank16) [inline services/data_deletion/clean_archived_data.py:13]: would you please put in a docstring which explains what this function, specifically to delete data which are 30 days old with also an archived state.

File: services/data_deletion/clean_archived_data.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    metadata = MetaData()
+    metadata.reflect(bind=database_engine)
+    print('deleting archived data...')
+
+    for table in reversed(metadata.sorted_tables):
+        try:
+            now = datetime.now()
+            delta = now - timedelta(days=30)
+            statement = table.delete().where(
+                and_(table.c.date_updated < delta, table.c.state == 'archived'))
```
Comment (MCFrank16) [inline services/data_deletion/clean_archived_data.py:22]: @mifeille after finishing the job, this is what celery.err.log is saying `sqlalchemy.exc.IntegrityError: (psycopg2.IntegrityError) update or delete on table "rooms" violates foreign key constraint "room_tags_room_id_fkey" on table "room_tags"
DETAIL:  Key (id)=(1) is still referenced from table "room_tags".
 [SQL: 'DELETE FROM rooms WHERE rooms.date_updated < %(date_updated_1)s AND rooms.state = %(state_1)s'] [parameters: {'date_updated_1': datetime.datetime(2019, 10, 5, 14, 31, 1, 378610), 'state_1': 'archived'}] (Background on this error at: http://sqlalche.me/e/gkpj)`

Comment (mifeille) [reply services/data_deletion/clean_archived_data.py:22]: There is a migration to include cascading deletion I guess that one on room tags was forgotten, I am going to add it.

File: services/data_deletion/clean_archived_data.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+@celery.task(name='clean_archived_data.delete_archived_data')
+def delete_archived_data():
+    """
+        This method deletes data that has been deleted for
+        more than 30 days
+        """
+    database_engine = create_engine(database_uri)
+    metadata = MetaData()
+    metadata.reflect(bind=database_engine)
+    print('deleting archived data...')
```
Comment (joshuaocero) [inline services/data_deletion/clean_archived_data.py]: Please remove this debug statement.

**Review by joshuaocero — CHANGES_REQUESTED** [review]
@mifeille this looks really good. Please remove the `print` statement so I can merge.

**Review by joshuaocero — APPROVED** [review]

**Pull request merged**